Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -12320,8 +12321,12 @@ void SpeculativeJIT::emitSwitchStringOnString(Node* node, SwitchData* data, GPRR
     
     std::sort(cases.begin(), cases.end());
     
-    emitBinarySwitchStringRecurse(
-        data, cases, 0, 0, cases.size(), string, lengthGPR, tempGPR, 0, false);
+    emitBinarySwitchStringRecurse(data, cases, 0, 0, cases.size(), string, lengthGPR, tempGPR, 0, false);
+
+    isRopeCases.link(this);
+    load32(Address(string, JSRopeString::offsetOfLength()), tempGPR);
+    sub32(TrustedImm32(unlinkedTable.minLength()), tempGPR);
+    branch32(Above, tempGPR, TrustedImm32(unlinkedTable.maxLength() - unlinkedTable.minLength()), data->fallThrough.block);
     
```

Comment:
Is this correct?  The Slow Case falls through into the Rope Case.  What happens when the input string is not a rope?

Reply:
Yes, this is correct. `emitBinarySwitchStringRecurse` will always end with jump to `data->fallThrough.block` of the switch cases. Thus, it does not go to here.